src/deweb/redirection.ts

```diff
@@ -64,24 +114,56 @@ export async function resolveDeweb(uri: string): Promise<string> {
     return defaultUrl
   }
 
+  // try to fetch the deweb info data from the current domain
   try {
     const response = await fetch(
       `${currentProtocol}//${currentHost}/__deweb_info`
     )
-    if (await isDewebInfoData(response)) {
+    if (await checkDewebInfoData(response, chainId)) {
       subdomains[0] = mnsDomain
-      return `${currentProtocol}//${subdomains.join('.')}${urlSearch}`
+      return `${currentProtocol}//${subdomains.join('.')}${urlSearch}${mnsHash}`
     }
     return defaultUrl
   } catch (error) {
+    // return default url if the current domain is not a Deweb provider
     return defaultUrl
   }
 }
 
-async function isDewebInfoData(response: Response): Promise<boolean> {
+/* 
+Check if the response correspond to a provider's __deweb_info endpoint response
+If the chainId parameter is not zero, we check if it corresponds to the chainId of the provider
+ */
+async function checkDewebInfoData(
```

Comment:
it is not clear what this function is doing reading the name.
Either the name can be improved, or the code refactored to put the chainId check outside this function